gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py

```diff
@@ -0,0 +1,229 @@
+import socket
+
+from neutron.common import exceptions
+from neutron.common import rpc as n_rpc
+from neutron.db import agents_db
+from neutron.db import agentschedulers_db
+from neutron import manager
+from neutron_vpnaas.db.vpn import vpn_validator
+from neutron_vpnaas.services.vpn.common import constants as const
+from neutron_vpnaas.services.vpn.common import topics
+from neutron_vpnaas.services.vpn.plugin import VPNPlugin
+from neutron_vpnaas.services.vpn import service_drivers
+from neutron_vpnaas.services.vpn.service_drivers import base_ipsec
+
+from oslo_log import log as logging
+import oslo_messaging
+
+LOG = logging.getLogger(__name__)
+
+BASE_VPN_VERSION = '1.0'
+
+
+class VPNAgentHostingServiceNotFound(exceptions.NeutronException):
+    message = _("VPN Agent hosting vpn service '%(vpnservice_id)s' not found")
+
+
+class VPNAgentNotFound(exceptions.NeutronException):
+    message = _("VPN Agent not found in agent_db")
+
+
+class VPNPluginExt(VPNPlugin, agentschedulers_db.AgentSchedulerDbMixin):
+    """
+    Extends the base VPN Plugin class to inherit agentdb too.
+    Required to get agent entry into the database.
+    """
+
+    def __init__(self):
+        super(VPNPluginExt, self).__init__()
+
+
+class OCIPsecVpnDriverCallBack(object):
```

Comment:
Change the class name to NFPIPsecVpnDriverCallBack.
Import base_ipsec.IPsecVpnDriverCallBack and extend that.